### PR TITLE
Update to clap >3.0.0

### PR DESCRIPTION
**Pull request by MatthiasZepper**
Clap, the package's Command Line Argument Parser, underwent some major overhaul of the syntax since version 0.1.2 BarcodEX-rs was published. 

While the `Cargo.lock` file ensures that the compilation still works for now, the build fails if just the current `Cargo.toml` file is considered:
 
```shell
rm Cargo.lock   
cargo clean
cargo build --release
```
This is relevant, since the Bioconda Build system disregards `Cargo.lock` files. [Hence, BarcodeEX-rs can't be added to Bioconda in its current form](https://www.biostars.org/p/9530801/#9530803).  

This PR seeks to update Clap with changing the least amount of code possible. Ultimately, a bigger rewrite might be advised, but for now only the required changes were made:  

- [replacing deprecated `Arg::about` in favor of `Arg::help`](https://github.com/clap-rs/clap/commit/7e899cd340c01b13388e7bb3649f4252924ad21f) 
- fixing compile error _Parser is imported here, but it is only a trait, without a derive macro_  using `features = [ "derive" ]` in `Cargo.toml `to enable the ability to derive.
- Additionally, `bio::io::fastq::Record` has no method `name()`, which was therefore replaced with `id()`.

### CODE REVIEW

**Comment by mlaszloffy** [comment]
Thank you very much @MatthiasZepper for your contribution!

**Review by mlaszloffy — APPROVED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
-clap ="3.0.0-beta.1"
+clap = { version = "^3.1", features = ["derive"]}
+derive = "*"
 flate2 = "^1.0"
 lazy_static = "^1.4"
 regex = "^1.3"
-serde = { version = "^1.0", features = ["derive"]}
 serde_json = "^1.0"
+
+[dependencies.serde]
```
Comment (slazicoicr) [inline Cargo.toml:16]: Why change to this format? The new `clap` line uses the old format and the official `sarde` documentation used the old format (https://serde.rs/derive.html)

File: src/main.rs [hunk 2]
```diff
@@ -448,61 +447,57 @@ fn write_stats(
     .unwrap();
 }
 
-#[derive(clap::Clap)]
+#[derive(Parser)]
 #[clap(
     version = "0.1",
     author = "Andre Masella <andre.masella@oicr.on.ca>",
-    about = "A package for extracting Unique Molecular Identifiers (UMIs) from single or paired read sequencing data"
+    //help = "A package for extracting Unique Molecular Identifiers (UMIs) from single or paired read sequencing data"
```
Comment (slazicoicr) [inline src/main.rs:454]: Why are the help commands commented out? Some are, some aren't.

File: Cargo.toml (hunk 1 above)
Comment (MatthiasZepper) [reply Cargo.toml:16]: I don't think that I edited this file by hand. I just used the `cargo` tool to update, add and remove dependencies.

**Comment by MatthiasZepper** [comment]
Admittedly, I completely forgot about this PR and I need to check what I did back then. Presumably, the most reasonable thing would be to close it and start over?

Back then, I had absolutely no clue about Rust and just tried to troubleshoot the code compile errors based on the compiler's suggestions. The reason was that we wanted to use the tool and since your toolchains are entirely based on Bioconda, we wanted to add it there. So my only goal was to change the code to the extent, that the automated Bioconda builds and deployment work.

Since the tool appeared unmaintained to us, we have since started our own development (mostly performed by a summer student at our facility) and I have also started to teach myself Rust. 

If you are still interested, I will take another look at this, but I am also fine with closing this PR and not following up on the issues.

**Comment by MatthiasZepper** [comment]
I am not very keen into looking into this old PR any more, and it seems the people at OICR aren't either, so best close once and for all.

**Pull request closed without merging**